### DISCUSSION TITLE
Deploying on Google App Engine (GAE), with .env*.local

### DISCUSSION THREAD

**Discussion by levanidev** [vercel/next.js#27968] [Help]
When deploying Next.js app on GAE, I noticed that .env*.local needed to be deployed with the production build in order to work.
I was adding all .env files to .gcloudignore and was wondering why it was not working.
Am I correct to assume that .env*.local files are used only when **next start/dev** is called?
Should I always deploy .env files? I read somewhere that it's not really a good idea to bake in .env files with deployment build because google cloud build keeps all the files.

What is the best practice here?

**Comment by wwwdepot** [comment]
You can define `env_variables` in the `app.yaml` file https://cloud.google.com/appengine/docs/standard/nodejs/config/appref#environment_variables

**Reply by levanidev** [reply]
usually app.yaml is committed to repo, so not sure if its a good idea to store secrets there

**Reply by wwwdepot** [reply]
You could extract **env_variables** to a separate yaml file and include it in your **app.yaml**.

**app.yaml**
```yaml
# include environment variables
includes:
- env_variables.yaml
```
**env_variables.yaml** (add this file to gitignore)
```yaml
env_variables:
  MY_SECRET_VAR: "value2"
```

**Reply by levanidev** [reply]
oh wow didn't know that. that is actually extremily helpful! thank you!

**Reply by levanidev** [reply]
Is this Yaml thing or App Engine supports it separately? can't seem to find docs on that
